crates/engine/src/tests.rs

```diff
@@ -272,3 +272,20 @@ fn ecdsa_recovery_with_secp256k1_crate() {
     // then
     assert_eq!(output, pubkey.serialize());
 }
+
+#[test]
+fn setting_getting_block_timestamp() {
+    // given
+    let mut engine = Engine::new();
+    let new_block_timestamp: u64 = 1000;
+    let output = &mut &mut get_buffer()[..];
+
+    // when
+    engine.set_block_timestamp(new_block_timestamp);
+    engine.block_timestamp(output);
```

Comment:
The test is pretty good and clearly readable. One thing: could you include testing `advance_block` in the test?

Reply:
Essentially, one might want to use the off_chain engine that way : 

```rust 
engine.set_block_time(new_block_time);
engine.advance_block();
engine.block_timestamp(); // Should return the previous block_timestamp + new_block_time
```

"Incomplete" might not be the best word here, but rather, "could have a complementary implementation".